connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -3507,6 +3513,47 @@ class SparkConnectPlanner(
         .build())
   }
 
+  private def handleCheckpointCommand(
+      checkpointCommand: CheckpointCommand,
+      responseObserver: StreamObserver[proto.ExecutePlanResponse]): Unit = {
+    val target = Dataset
+      .ofRows(session, transformRelation(checkpointCommand.getRelation))
+    val checkpointed = if (checkpointCommand.hasLocal && checkpointCommand.hasEager) {
```

Comment:
You could also increase the visibility of `Dataset.checkpoint(eager: Boolean, reliableCheckpoint: Boolean)`.

Reply:
I just did it locally but I think it's actually better to keep just as is .. I think the current one is easier to read .. `Dataset.checkpoint(eager: Boolean, reliableCheckpoint: Boolean)` is private as well.